service/history/replication/cache.go

```diff
@@ -98,14 +109,22 @@ func (c *Cache) Put(task *types.ReplicationTask) error {
 		return nil
 	}
 
+	taskSize := task.ByteSize()
+
+	// Check for full cache
+	if (len(c.order) >= c.maxCount()) || (c.currSize+taskSize) >= uint64(c.maxSize()) {
```

Comment:
A question: Do we have a unit test to cover the currentSize hitting it's max value, to ensure it handles integer overflow correctly? 

Do you mind adding a guard to ensure that the combined value of currSize can't overflow when added iwth tasksize, thereby bypassing the check? (I guess if c.currSize == uint64 maxInt + 1, does that bypass?)